### PR TITLE
Add C implementation for `@stdlib/math/base/special/dirac-delta`

### LINKED ISSUE #675: [RFC]: Add C implementation for `@stdlib/math/base/special/dirac-delta`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/dirac-delta`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/dirac-delta)
```c
double stdlib_base_dirac_delta( const double x )
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #675 .

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

**Comment by Pranavchiku** [comment]
All tests and benchmarks passed, and this PR can be reviewed !

File: lib/node_modules/@stdlib/math/base/special/dirac-delta/benchmark/c/native/benchmark.c [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/**
+* Benchmark `diracdelta`.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/dirac-delta/benchmark/c/native/benchmark.c]:
```suggestion
* Benchmark `dirac_delta`.
```

File: lib/node_modules/@stdlib/math/base/special/dirac-delta/benchmark/c/native/benchmark.c [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+*/
+#include "stdlib/math/base/special/dirac_delta.h"
+#include <stdlib.h>
+#include <stdio.h>
+#include <math.h>
+#include <time.h>
+#include <sys/time.h>
+
+#define NAME "dirac-delta"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/dirac-delta/benchmark/c/native/benchmark.c]:
```suggestion
#define NAME "dirac_delta"
```

File: lib/node_modules/@stdlib/math/base/special/dirac-delta/examples/c/example.c [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+int main() {
+    double x[] = { -1.0, -0.5, 0.0, 0.5, 1.0, 3.14, 2.0 };
+    double v;
+    int i;
+
+    for ( i = 0; i < 7; i++ ) {
+        v = stdlib_base_dirac_delta( x[ i ] );
+        printf( "dirac(%lf) = %lf\n", x[ i ], v );
+    }
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/dirac-delta/examples/c/example.c]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/dirac-delta/README.md [hunk 4]
```diff
@@ -101,6 +101,97 @@ for ( i = 0; i < x.length; i++ ) {
...
+int main() {
+    double x[] = { -1.0, -0.5, 0.0, 0.5, 1.0, 3.14, 2.0 };
+    double v;
+    int i;
+
+    for ( i = 0; i < 7; i++ ) {
+        v = stdlib_base_dirac_delta( x[ i ] );
+        printf( "dirac(%lf) = %lf\n", x[ i ], v );
+    }
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/dirac-delta/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/dirac-delta/include/stdlib/math/base/special/dirac_delta.h [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+/*
+* If C++, prevent name mangling so that the compiler emits a binary file having undecorated names, thus mirroring the behavior of a C compiler.
+*/
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+* Evaluates the Dirac Delta function.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/dirac-delta/include/stdlib/math/base/special/dirac_delta.h]:
```suggestion
* Evaluates the Dirac delta function.
```

File: lib/node_modules/@stdlib/math/base/special/dirac-delta/lib/native.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Evaluates the Dirac delta function.
+*
+* @param {number} x - input value
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/dirac-delta/lib/native.js:32]:
```suggestion
* @private
* @param {number} x - input value
```

File: lib/node_modules/@stdlib/math/base/special/dirac-delta/src/main.c [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/dirac_delta.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/constants/float64/pinf.h"
+/**
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/dirac-delta/src/main.c:23]:
```suggestion

/**
```

File: lib/node_modules/@stdlib/math/base/special/dirac-delta/src/main.c [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+* Evaluates the Dirac Delta function.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/dirac-delta/src/main.c]:
```suggestion
* Evaluates the Dirac delta function.
```

File: lib/node_modules/@stdlib/math/base/special/dirac-delta/src/main.c [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+*
+* @param x    input value/
+* @return	  output value.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/dirac-delta/src/main.c]:
```suggestion
* @param x    input value
* @return	  output value
```

File: lib/node_modules/@stdlib/math/base/special/dirac-delta/src/main.c [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+*
+* @example
+* double x = 3.14;
+*
+* double out = stdlib_base_rad2deg( x );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/dirac-delta/src/main.c]:
```suggestion
* double out = stdlib_base_dirac_delta( 3.14 );
```

File: lib/node_modules/@stdlib/math/base/special/dirac-delta/README.md [hunk 11]
```diff
@@ -101,6 +101,96 @@ for ( i = 0; i < x.length; i++ ) {
...
+<section class="examples">
+
+### Examples
+
+```c
+#include "stdlib/math/base/special/dirac_delta.h"
+#include <stdlib.h>
+#include <stdio.h>
+
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/dirac-delta/README.md]:
```suggestion
```

**Review by kgryte — APPROVED** [review]

**Pull request merged**